pkg/workload/workload.go

```diff
@@ -495,13 +535,47 @@ func SetEvictedCondition(w *kueue.Workload, reason string, message string) {
 	apimeta.SetStatusCondition(&w.Status.Conditions, condition)
 }
 
+// PropagateResouceRequests synchronizes w.Status.ResourceRequests to
+// with info.TotalRequests and returns true if w was updated
+func PropagateResourceRequests(w *kueue.Workload, info *Info) bool {
+	if len(w.Status.DesiredResources) == len(info.TotalRequests) {
+		match := true
+		for idx := range w.Status.DesiredResources {
+			if w.Status.DesiredResources[idx].Name != info.TotalRequests[idx].Name ||
+				!maps.Equal(w.Status.DesiredResources[idx].Resources, info.TotalRequests[idx].Requests.ToResourceList()) {
+				match = false
+				break
+			}
+		}
+		if match {
+			return false
+		}
+	}
+
+	res := make([]kueue.PodSetRequest, len(info.TotalRequests))
+	for idx := range info.TotalRequests {
+		res[idx].Name = info.TotalRequests[idx].Name
+		res[idx].Resources = info.TotalRequests[idx].Requests.ToResourceList()
+	}
+	w.Status.DesiredResources = res
+	return true
+}
+
 // AdmissionStatusPatch creates a new object based on the input workload that contains
 // the admission and related conditions. The object can be used in Server-Side-Apply.
 // If strict is true, resourceVersion will be part of the patch.
 func AdmissionStatusPatch(w *kueue.Workload, strict bool) *kueue.Workload {
 	wlCopy := BaseSSAWorkload(w)
 	wlCopy.Status.Admission = w.Status.Admission.DeepCopy()
 	wlCopy.Status.RequeueState = w.Status.RequeueState.DeepCopy()
+	if wlCopy.Status.Admission != nil && len(w.Status.DesiredResources) > 0 {
```

Comment:
Do we need this check "len(w.Status.DesiredResources) > 0"? Can we just do:
```golang
if wlCopy.Status.Admission != nil {
   wlCopy.Status.DesiredResources = []kueue.PodSetRequest{} // or maybe even nil would work?
} else {
   slices.copy(wlCopy.Status.DesiredResources, w.Status.DesiredResources)
}
```

Reply:
You are correct; the check for` len > 0` is not needed (since iterating over a length 0 array does nothing).
To be consistent with the rest of the copying done by this function, I left an explicit loop that uses `DeepCopy()`  since `slices.Clone` only does a shallow copy.

Reply:
sgtm, thanks